server/routes/contacts/create/check-answers/createContactCheckAnswersController.ts

```diff
@@ -32,6 +32,11 @@ export default class CreateContactCheckAnswersController implements PageHandler
     await this.contactService
       .createContact(journey, user)
       .then(() => delete req.session.createContactJourneys[journeyId])
-    res.redirect(`/prisoner/${prisonerNumber}/contacts/create/success`)
+
+    if (journey.returnPoint.type === 'MANAGE_PRISONER_CONTACTS') {
+      res.redirect(journey.returnPoint.url)
+    } else {
+      res.redirect(`/prisoner/${prisonerNumber}/contacts/create/success`)
+    }
```

Comment:
I'll also test the returning from the end of one mini-journey back into the journey it came from and that the original  journey URL including ID is still in the `returnPoint` of the new mini-journey. Seems like a reasonable way to deal with this, but lets see in practice.

Reply:
I was planning to remove the success page but that had the side effect of breaking every single test so deferred. I think the default should actually be straight to prisoner contacts list page but there is no entry point for that currently.